python/paddle/fluid/optimizer.py

```diff
@@ -475,7 +475,7 @@ def backward(self,
             for param in parameters:
                 if not param.trainable:
                     continue
-                if param._ivar._grad_ivar() is not None:
+                if param._ivar._grad_ivar_is_initialized():
```

Comment:
Better to add a test to check if this works with the neural network with only part of its weight initialized